### DISCUSSION TITLE
SSH-AGENT use .ssh/id_rsa

### DISCUSSION THREAD

**Discussion by SergejGuruliov** [Nitrux/nitrux-bug-tracker#196] [General]
# Issue Description
SSH-AGENT

~ ❯ ssh-agent                                                                                                                                                 
SSH_AUTH_SOCK=/tmp/ssh-CSipqbrTwLnR/agent.23478; export SSH_AUTH_SOCK;
SSH_AGENT_PID=23479; export SSH_AGENT_PID;
echo Agent pid 23479;
~ ❯ ssh-add                                                                                                                                              
Could not open a connection to your authentication agent.

**Comment by UriHerrera** [comment]
Add context. I have no idea what's going on here or what you're trying to do.

**Reply by SergejGuruliov** [reply]
What is ssh-agent?

ssh-agent is a background process (daemon) used to manage and store SSH private keys. It is part of the SSH (Secure Shell) suite of tools designed for secure connections to remote systems.

The main purpose of ssh-agent is to:

    Store private keys: It keeps private SSH keys in memory so that you don't need to repeatedly enter their passphrase.
    Facilitate authentication: It allows other programs (like ssh, scp, or git) to access these keys for authentication with remote servers.

Why is ssh-agent useful?

Without ssh-agent, you would need to manually enter the passphrase for your private key every time you use it. ssh-agent simplifies this process by:

    Improving security: The keys are stored in RAM (not on disk), making them harder to steal.
    Enhancing convenience: You only need to enter your passphrase once per session, and ssh-agent handles authentication automatically after that.

**Reply by SergejGuruliov** [reply]
So after eval. It's work but only for current terminal session. Not work globally with apps like git, sublime, vs code ...
~ ❯ eval "$(ssh-agent -s)"                                                                                                                      
Agent pid 28416
~ ❯ ssh-add                                                                                                                                              
Enter passphrase for /home/user/.ssh/id_rsa: 
Identity added: /home/user/.ssh/id_rsa (/home/user/.ssh/id_rsa)

**Reply by SergejGuruliov** [reply]
It also works with Nitrux 2.5.1.
